lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -187,18 +187,19 @@ def test_unpack(
     ):
         """Test to check the unpacking functionality of the FileTree"""
         source_tarball, source_md5, md5 = tarball
-        tarball_name = source_tarball.name[0 : -len(".tar.xz")]
 
-        def mock_run(args, cwd, stdin, stderr, stdout):
+        def mock_run(args, cwd, stdin, stderr, stdout, universal_newlines):
```

Comment:
Somehow, and I think a bit irrationally, the `universal_newlines` makes me uncomfortable. In the real `run` I'm pretty sure this is a `kwarg` and it's definitely optional, which means that (at least) the signature ought to be `universal_newlines = False` if not `universal_newlines: bool = False`.

On the other hand ... you're already failing if it's not specified as `True` (technically, "if it's not truth-y"), so having Python fail if it's not specified at all doesn't hurt.

I wish GitHub allowed me to make "non-binding commentary" that wouldn't count against a merge. Since I'm not insisting on any change here, I guess I'll just pre-resolve the comment. 😆

Reply:
I'll second Dave's comments, here.  (Also, non-binding, so I won't "unresolve" this conversation....)

This mock probably _should_ look more like this:
```
        def mock_run(args, *, **kwargs):
            ...
            assert kwargs["stdin"] == subprocess.DEVNULL
            ...
```
That way, if the `stdin` parameter has the wrong value, the assertion will fail, and if the argument is omitted we'll get an exception when we look for it, which will also cause the test to fail.  (We could assert that `"stdin"` is `in kwargs`, but I don't think it's worth it.)

I don't know what will happen if the CUT omits one of the mock functions arguments -- it'll probably raise an exception at the call site...which will, admittedly, make the test fail (probably...), but I don't really want to count on that.  Moreover, I'm also worried that, if the CUT is changed to specify the (keyword!) arguments in a different order, I'm not sure whether Python will "connect them up" properly or not, and that could result in weird behaviors which might be hard to debug.